testdata/p4_16_samples_outputs/synth-action.p4-stderr

```diff
@@ -0,0 +1,3 @@
+synth-action.p4(22): [--Wwarn=mismatch] warning: 4294967302: value does not fit in 32 bits
+            x = x - 6;
```

Comment:
This one is truly strange and makes me scratch my head as to where the negative value is coming from?

```
control c(inout bit<32> x) {
    apply {
        x = 10;
        if (x == 10) {
            x = x + 2;
            x = x - 6;
        } else {
            x = x << 2;
        }
    }
}
```

Reply:
If you put a breakpoint in the `warning` function, you can figure out which `IR::Node` that value is coming from.  You can then put a breakpoint in IR::Node::traceCondition with a condition `id == `... to see where that specific IR::Node gets created...

Reply:
Okay, this is because we evaluate

```
x = x - 6;
```

as

```
x= x + (-6);
```

And `IR::Constant` thinks -6 is is a large constant ...